ansible/roles/esdc-mgmt/templates/esdc/es-post-deploy.sh.j2

```diff
@@ -68,6 +68,28 @@ chown root:root /etc/pki/tls/certs/server.pem
 chmod 600 /etc/pki/tls/certs/server.pem
 chmod 600 /etc/pki/tls/certs/server.key
 
+# setup mgmt services SSL proxy
+SVC_CERT_COMBINED_B64="$(mdata-get org.erigones:erigonesd_ssl_cert 2> /dev/null)"
+CFGDB_IP="$(mdata-get org.erigones:cfgdb_ip 2> /dev/null)"
+SVC_CERT_FILE="{{ esdc_mgmt.svc_cert_file }}"
+
+if [[ -z "${CFGDB_IP}" ]]; then
+	CFGDB_IP="127.0.0.1"
+fi
+
+if [[ -n "${SVC_CERT_COMBINED_B64}" ]]; then
+	# certificate exists, write it to file
+	echo "${SVC_CERT_COMBINED_B64}" | /usr/bin/base64 -d > "${SVC_CERT_FILE}"
+else
+	# cert was not provided
+	# we have to generate the cert and push it to mdata
+	openssl req -x509 -sha256 -nodes -days 10000 -subj '/CN=*.*/O=Danube Cloud/' -newkey rsa:8192 -keyout "${SVC_CERT_FILE}" -out "${SVC_CERT_FILE}"
+	mdata-put "org.erigones:erigonesd_ssl_cert" "$(sed -ne "/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p" ${SVC_CERT_FILE} | /usr/bin/base64)"
+fi
+
+chmod 600 "${SVC_CERT_FILE}"
+sed -i'' -e "s/%CFGDB_IP%/${CFGDB_IP}/g" /etc/haproxy/haproxy.cfg
```

Comment:
Please see comment haproxy.cfg.j2. We can change this so it is replaced only if the cfgdb_ip metadata is found but the search pattern will be different.